### PR TITLE
SOV-3237: Proposal creation - Proposal Overview

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-3237

### CODE REVIEW

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+    </div>
+  );
+};
+
+export const formatProposalText = (
+  title: string,
+  discussionUrl: string,
+  summary: string,
+  proposalText: string,
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx]: Can just accept entire form

```suggestion
{ title, link, summary, text }: ProposalDetails
```

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+) => {
+  const formattedTitle = `"${title}"`;
+  const formattedDiscussionUrl = `"${discussionUrl}"`;
+  const formattedSummary = `"${summary}"`;
+  const formattedProposalText = `"${proposalText}"`;
+  const proposalFormatText = `${formattedTitle}\n${formattedDiscussionUrl}\n${formattedSummary}\n---\n${formattedProposalText}`;
+  return proposalFormatText;
+};
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx]: `"` is not needed, Light just used them in the PRD to indicate it's text that needs to be variable.

```suggestion
) => `${title}\n${link}\n${summary}\n---\n${text}`;
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/NewProposalForm.tsx [hunk 3]
```diff
@@ -43,14 +44,23 @@ export const NewProposalForm: FC = () => {
...
 
   const handleSubmit = useCallback(async () => {
+    const proposalFormatText = formatProposalText(
+      details.title,
+      details.link,
+      details.summary,
+      details.text,
+    );
+
+    console.log('formatted data', proposalFormatText);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/NewProposalForm.tsx]:
```suggestion

```

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/NewProposalForm.tsx [hunk 4]
```diff
@@ -43,14 +44,23 @@ export const NewProposalForm: FC = () => {
   }, [setStep]);
 
   const handleSubmit = useCallback(async () => {
+    const proposalFormatText = formatProposalText(
+      details.title,
+      details.link,
+      details.summary,
+      details.text,
```
Comment (creed-victor) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/NewProposalForm.tsx]:
```suggestion
      details
```

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const maxCharactersLength = 140;
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.constants.ts]: Split it into 2 constants - `MAXIMUM_TITLE_LENGTH` and `MAXIMUM_SUMMARY_LENGTH`.

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.constants.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
+export const maxDiscussionUrlLength = 500;
+export const maxProposalTextLength = 30000;
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.constants.ts]:
```suggestion
export const MAXIMUM_DISCUSSION_URL_LENGTH = 500;
export const MAXIMUM_TEXT_LENGTH = 30000;
```

File: apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+) => {
+  return (
+    <div className="flex items-center justify-between w-full">
+      <div>{labelText}</div>
+      <div className="text-gray-30">
+        {value.length}/{maxLength}
+      </div>
+    </div>
+  );
+};
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/ProposalDataForm/ProposalDataForm.utils.tsx]:
```suggestion
export const generateFormGroupLabel = (
  labelText: string,
  value: string,
  maxLength: number | string,
) => (
    <div className="flex items-center justify-between w-full">
      <div>{labelText}</div>
      <div className="text-gray-30">
        {value.length}/{maxLength}
      </div>
    </div>
  );
```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by creed-victor — APPROVED** [review]

**Pull request merged**